### DISCUSSION TITLE
Change Style each Row not Working with "getTableRecordClassesUsing()"

### DISCUSSION THREAD

**Discussion by DasPocky** [filamentphp/filament#6641] [Help] [answered]
Hi,

I read in the documentation that you can style the individual rows if a certain value applies. I have now tried a few things, but the row is not styled based on the status. (https://filamentphp.com/docs/2.x/tables/getting-started#record-classes). I found a similar post already here in the discussion, but this one didn't get me anywhere.

OrderResource.php

```php

use use Closure;
use App\Models\Order;

protected function getTableRecordClassesUsing(): ?Closure
{
    Log::info('getTableRecordClassesUsing() called');

    return fn (Order $record) => match ($record->status) {
        'OrderPlanned' => 'border-l-2 border-orange-500',
        'LoadingCompleted' => 'border-l-2 border-green-500',
        default => null,
    };
}
```

tailwind.config.js

```js
const colors = require('tailwindcss/colors')

/** @type {import('tailwindcss').Config} */
module.exports = {
    content: [
        "./resources/**/*.blade.php",
        "./resources/**/*.js",
        "./resources/**/*.vue",
        "./vendor/filament/**/*.blade.php",
        "./app/Filament/**/*.php"
    ],
    darkMode: 'class',
    theme: {
        screens: {
            'sm': '564px',
            'md': '768px',
            'lg': '1024px',
            'xl': '1280px',
            '2xl': '1536px'
        },
        extend: {
            colors: {
                danger: colors.red,
                primary: colors.blue,
                success: colors.green,
                warning: colors.yellow,
            },
        },
    },
    plugins: [
        require('@tailwindcss/forms'),
        require('@tailwindcss/typography'),
    ],
}
```

Normally, the styling should change based on the status, but it has no effect. To check if the function is called at all, I had a log entry created. The log remained empty. 

Sorry for my bad English, I am not so good at writing in English, so I got a translator to help me. I still hope my problem will be clear.

Does anyone have a solution I could try?

Greeting, Björn

**Comment by danharrin** [comment, marked as answer]
Ensure `getTableRecordClassesUsing()` is on the list page class, not the resource class

**Reply by DasPocky** [reply]
Okay, error number one was that I didn't write it into the ListOrders class, but I tried it shortly and it didn't work.

Now I have done it. Thank you very much for the idea.

```php
use Closure;
use Illuminate\Database\Eloquent\Model;

protected function getTableRecordClassesUsing(): ?Closure
{
    return fn (Model $record) => match ($record->status) {
        'OrderPlanned' => 'border-l-4 border-l-orange-500',
        'LoadingCompleted' => 'border-l-4 border-l-blue-500',
        'UnloadingCompleted' => 'border-l-4 border-l-green-500',
        'WithoutCleaning' => 'border-l-4 border-l-yellow-500',
        'CleaningCompleted' => 'border-l-4 border-l-purple-500',
        'OrderCompleted' => 'border-l-4 border-l-red-500',
        default => 'border-l-4 border-l-gray-500',
    };
}
```

**Comment by tnt-ylli** [comment]
How does it work for you, I have absolutely the same thing! Not working at all

**Reply by leandrocfe** [reply]
Are you using Filament v2?

V3 is different
https://filamentphp.com/docs/3.x/tables/advanced#custom-row-classes

**Reply by tnt-ylli** [reply]
Yes , I'm on V3, firstly i tried this approach, but did not worked. Maybe Im missing out something, but one for sure ,I have added "./app/Filament/**/*.php" inside tailwind config , and using the recordClasses as described in documentation.

**Reply by leandrocfe** [reply]
It was supposed to work. You could open a new help topic and share the code that you are using.
